### DISCUSSION TITLE
Sub query doesn't work in Trend Metric

### DISCUSSION THREAD

**Discussion by ysfkaya** [laravel/nova-issues#3189] [Q&A] [answered]
- Laravel Version: ^7.0
- Nova Version: ^3.0
- PHP Version: ^7.4
- Database Driver & Version: Mysql ^5.7
- Operating System and Version: MacOS Catalina 10.15.5
- Browser type and version: Mozilla Firefox

### Description:

When I trying the custom eloquent query with using sub query I'm getting a `Column not found` error from mysql.

<img width="880" alt="Ekran Resmi 2021-01-22 14 42 15" src="https://user-images.githubusercontent.com/16710972/105486724-06a6c100-5cc0-11eb-9ccd-4a84f230e05c.png">

So I think, Nova recalling `select`query in trend metric `aggregate` method and so my sub query is not applying.

<img width="1196" alt="Ekran Resmi 2021-01-22 14 43 07" src="https://user-images.githubusercontent.com/16710972/105486784-25a55300-5cc0-11eb-9350-1c0104aaf9e1.png">

Is there any way to prevent this?

**Comment by crynobone** [comment, marked as answer]
For such complex query I would suggest overriding the `aggregate()` to best meet your requirement.

**Comment by ysfkaya** [comment]
Well @crynobone, I thought it was a general problem thanks.
